src/components/Login.jsx

```diff
@@ -1,31 +1,71 @@
 import PropTypes from 'prop-types'
 import React from 'react'
+import { withRouter } from 'react-router'
+
 import { isClient } from '../lib'
 import UserEdit from '../containers/UserEdit'
 
-const Login = ({ location }) => (
-  <div>
-    {isClient() ? '' /*window.AuthService.login(location.query.nextUrl)*/ : ''}
-    <form action="/login-callback" method="POST">
-      <input type="hidden" name="nextUrl" value={location.query.nextUrl} />
-      <div>
-        <label>email:</label>
-        <input type="text" name="email"/><br/>
-      </div>
-      <div>
-	<label>Password:</label>
-	<input type="password" name="password"/>
-      </div>
+class Login extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      active: 'login'
+    }
+  }
+
+  handleClick = e => {
+    this.setState({ active: e.target.name })
+  }
+
+  render() {
+    const auth0Strategy =
+      window.PASSPORT_STRATEGY === 'auth0' ||
+      window.PASSPORT_STRATEGY === ''
+    const auth0Login = isClient() && auth0Strategy
+
+    const { location, router } = this.props
+
+    return (
       <div>
-	<input type="submit" value="Submit"/>
-      </div>
-    </form>
-    <UserEdit  />
-  </div>
-)
+        {/* Use auth0 */}
+        {auth0Login && window.AuthService.login(location.query.nextUrl)}
+
+        {/* Show UserEdit component configured for login / signup */}
+        <section>
+          <button
+            type='button'
+            name='login'
+            onClick={this.handleClick}
+            disabled={this.state.active === 'login'}
+          >
+            Log in
+          </button>
+          <button
+            type='button'
+            name='signup'
+            onClick={this.handleClick}
+            disabled={this.state.active === 'signup'}
+          >
+            Sign up
+          </button>
+        </section >
+
+        {!auth0Login &&
+          <UserEdit
+            authType={this.state.active}
+            saveLabel={this.state.active === 'login' ? 'Log in' : 'Sign up'}
+            router={router}
+            location={location}
```

Comment:
for `signup` state we need to test for and pass an invite code (either an invite table key or an org uuid).  If neither is present, then it should just show login without an option to signup.

(I'll comment further down, but we also need to validate this on the server and send through the UserEdit form)

Reply:
Should I also do the opposite, only show signup if there is an invite code?

Reply:
No, an invite is also to a new campaign eg for dyn assignment so often invites/uuid links will be clicked by existing users too